### ISSUE TITLE
Why are the test results of RDN model not good?

### ISSUE THREAD

**Issue by WenwuZhang** [thstkdgus35/EDSR-PyTorch#86]
I trained the model of rdn*4, the training results are good about 29.70 psnr, but when I test with the best model in benchmark, the results are bad about 12.00 psnr.  Similar to the following:
python3.6 main.py --scale 2 --save RDN_D16C8G64_BIx2 --model RDN --epochs 200 --batch_size 16 --patch_size 64 --reset
Making model...
Preparing loss function:
1.000 * L1
[Epoch 1]	Learning rate: 1.00e-4
[1600/16000]	[L1: 18.7452]	26.5+0.3s
[3200/16000]	[L1: 13.0720]	26.6+0.0s
[4800/16000]	[L1: 10.8100]	26.7+0.0s
[6400/16000]	[L1: 9.5034]	26.9+0.0s
[8000/16000]	[L1: 8.6019]	27.0+0.0s
[9600/16000]	[L1: 8.0266]	27.3+0.0s
[11200/16000]	[L1: 7.5577]	27.3+0.0s
[12800/16000]	[L1: 7.1941]	27.5+0.0s
[14400/16000]	[L1: 6.8882]	27.5+0.0s
[16000/16000]	[L1: 6.6593]	27.4+0.0s

Evaluation:
100%|███████████████████████████████████████████| 10/10 [00:31<00:00,  3.19s/it]
[DIV2K x2]	PSNR: 32.918 (Best: 32.918 @epoch 1)
Forward: 31.72s

Saving...
Total: 32.12s

python main.py --data_test Set5 --data_range 801-900 --scale 2 --pre_train ../experiment/RDN_D16C8G64_BIx2/model/model_best.pt --test_only --self_ensemble
Making model...
Load the model from ../experiment/RDN_D16C8G64_BIx2/model/model_best.pt

Evaluation:
100%|█████████████████████████████████████████████| 5/5 [00:00<00:00,  5.14it/s]
[Set5 x2]	PSNR: 11.440 (Best: 11.440 @epoch 1)
Forward: 0.61s

Saving...
Total: 0.61s

Even if I train 200 epochs, the results are still the same.Could you give me some advice? Much thanks!

**Comment by moleaf** [comment]
Hi, I met the same question like you, have you solved it and how ?

**Comment by WenwuZhang** [comment]
Just specify the default model as "RDN" when you test发自我的iPhone------------------ Original ------------------From: Lainey❤  <notifications@github.com>Date: Tue,Nov 20,2018 11:09 AMTo: thstkdgus35/EDSR-PyTorch <EDSR-PyTorch@noreply.github.com>Cc: WenwuZhang <zhangwenwu@hrbeu.edu.cn>, State change <state_change@noreply.github.com>Subject: Re: [thstkdgus35/EDSR-PyTorch] Why are the test results of RDN modelnot good? (#86)Hi, I met the same question like you, have you solved it and how ?

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/thstkdgus35/EDSR-PyTorch","title":"thstkdgus35/EDSR-PyTorch","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/thstkdgus35/EDSR-PyTorch"}},"updates":{"snippets":[{"icon":"PERSON","message":"@moleaf in #86: Hi, I met the same question like you, have you solved it and how ?"}],"action":{"name":"View Issue","url":"https://github.com/thstkdgus35/EDSR-PyTorch/issues/86#issuecomment-440121556"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/thstkdgus35/EDSR-PyTorch/issues/86#issuecomment-440121556",
"url": "https://github.com/thstkdgus35/EDSR-PyTorch/issues/86#issuecomment-440121556",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [thstkdgus35/EDSR-PyTorch] Why are the test results of RDN model not good? (#86)",
"sections": [
{
"text": "",
"activityTitle": "**Lainey❤**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@moleaf",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"thstkdgus35/EDSR-PyTorch\",\n\"issueId\": 86,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/thstkdgus35/EDSR-PyTorch/issues/86#issuecomment-440121556"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 412508375\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by moleaf** [comment]
Hi, have you reproduce the results reported in the paper using this code? If you have, could you please share me with your training settings? For Set5x2, I got the PSNR(38.167) in epoch 600, not reaching the reported value in paper(38.24).

**Comment by yangbincheng** [comment]
> Hi, have you reproduce the results reported in the paper using this code? If you have, could you please share me with your training settings? For Set5x2, I got the PSNR(38.167) in epoch 600, not reaching the reported value in paper(38.24).

Hi, moleaf! I want to reproduce the results in the paper, but the x2 model seems diverge. The best result is achieved about 30 epochs, the PSNR is 37.41% for Set5 x2, after 500+ epochs, the PSNR drops to 36.20%. I use Python3.7 and Pytorch 1.0.1. May you please share me with your training settings and software environment configuration?
